components/renku_data_services/data_connectors/orm.py

```diff
@@ -0,0 +1,136 @@
+"""SQLAlchemy schemas for the data connectors database."""
+
+from datetime import datetime
+from typing import TYPE_CHECKING, Any
+
+from sqlalchemy import JSON, Boolean, DateTime, ForeignKey, MetaData, String, func
+from sqlalchemy.dialects.postgresql import ARRAY, JSONB
+from sqlalchemy.orm import DeclarativeBase, Mapped, MappedAsDataclass, mapped_column, relationship
+from sqlalchemy.schema import Index
+from ulid import ULID
+
+from renku_data_services.authz import models as authz_models
+from renku_data_services.base_orm.registry import COMMON_ORM_REGISTRY
+from renku_data_services.data_connectors import models
+from renku_data_services.data_connectors.apispec import Visibility
```

Comment:
nitpick: I think using `from renku_data_services.authz.models import Visibility` is cleaner here, then there's no dependency here nor in db.py on the apispec. (I think there's other places in the code base that have this issue as well).

Reply:
I will defer this to later because using the `Visibility` from `renku_data_services.authz.models` causes the following error:

```
asyncpg.exceptions.InvalidTextRepresentationError: invalid input value for enum visibility: "PUBLIC"
```

Reply:
ah because that one is upper case. It's a bit weird that it takes the enum entry name instead of the string value. Maybe if that one was `StrEnum` like we use in other places it'd work.
Doing it later is fine, I think we need to clean this up in general.